library/Leads/Exporter/AbstractExcelExporter.php

```diff
@@ -52,16 +52,26 @@ public function export($config, $ids = null)
      */
     protected function exportWithFormat($config, $ids, $format)
     {
+        $actTime = time();
+        
+        $lastExportDate = null;
+        if ($config->onlyExportSinceLastExportDate && \Validator::isDatim($config->lastExportDate))
+        {
+          $lastExportDate = $config->lastExportDate;
+        }
+        
         $dataCollector = $this->prepareDefaultDataCollector($config, $ids);
-        $reader = new ArrayReader($dataCollector->getExportData());
+        $reader = new ArrayReader($dataCollector->getExportData($lastExportDate));
 
         if ($config->headerFields) {
             $reader->setHeaderFields($this->prepareDefaultHeaderFields($config, $dataCollector));
         }
 
         $row = new Row($config, $this->prepareDefaultExportConfig($config, $dataCollector));
 
-
+        \Database::getInstance()->prepare('UPDATE tl_lead_export SET lastExportDate = ? WHERE id = ?')
```

Comment:
Are you intentionally updating the date even if the flag is not enabled?
Also, if the export breaks for any reason, the date would still be updated (because it's too early)?